src/index.ts

```diff
@@ -48,7 +50,7 @@ export default function (input: Element, options?: Partial<Options>) {
 
   config = {...defaults, ...options}
 
-  rootDocument = findRootDocument(config.root)
+  rootDocument  = config.rootDocument !== undefined ? config.rootDocument as Element : findRootDocument(config.root)
```

Comment:
Better to setup correct rootDocument inside findRootDocument. See https://github.com/antonmedv/finder/pull/14 for more details on why this was added in the first place.

Reply:
The only thing I can see of doing here for this is to just always use the config.root as the root document as it defaults to the `document.body` anyway. Is there a reason you are using the `ownerDocument`instead of just the `document.body` or the element passed in?

All the functions you are calling on `rootDocument` are available on Elements as well.

Reply:
Actually. Just using the config.root as the `rootDocument` might solve this problem and #14 if i understand #14 correctly. I will push the change I think makes sense and ill see if I can set up a test with an iframe too to see if it works.